[9.0][MIG] Reports ported: VAT Listing and VAT Intra

robinkeunen opened the pull request:
Migration from `10.0/l10n_be_vat_reports` to 9.0.

Based on #57 and #74

andreparames committed:
Reports ported: VAT Listing and VAT Intra

andreparames committed:
[FIX] Correctly calculate turnover and vat_amount

This commit fixes two problems with the calculation:

- The turnover tax was being fetched from the wrong field
- Move lines with taxes that had more than one tag where being
summed more than once
Use -balance for turnover and tax

andreparames committed:
[FIX] vat reports: Create XML records in current module

They were being created in l10n_be

andreparames committed:
vat reports: add test

andreparames committed:
vat reports: improve backward compatibility

andreparames committed:
vat reports: upgrade sequence code

andreparames committed:
vat reports: fixing menu references

andreparames committed:
vat reports: fix UI problems in VAT intra report

andreparames committed:
vat reports: improved intra report

andreparames committed:
vat reports: fix encoding & remove pointless code

andreparames committed:
vat reports: small corrections

sbidoul committed:
[ADD] setup.py

robinkeunen committed:
[MIG] l10n_be vat reports

robinkeunen committed:
[REM] l10n_be_vat_reports: cleanup unused reports

robinkeunen committed:
[MIG] l10n_be_vat_reports: rename manifest

tbaden commented on l10n_be_vat_reports/wizard/l10n_be_vat_intra_view.xml:
```diff
@@ -0,0 +1,69 @@
+            </field>
+        </record>
+
+        <record id="view_vat_intra_save" model="ir.ui.view">
```
optional:
id pattern

> For a view: <model_name>_view_<view_type>, where view_type is kanban, form, tree, search, ...

tbaden commented on l10n_be_vat_reports/wizard/l10n_be_vat_intra_view.xml:
```diff
@@ -0,0 +1,69 @@
+<openerp>
+    <data>
+
+        <record id="view_vat_intra" model="ir.ui.view">
```
optional:
id pattern

> For a view: <model_name>_view_<view_type>, where view_type is kanban, form, tree, search, ...

tbaden commented on l10n_be_vat_reports/wizard/l10n_be_partner_vat_listing.xml:
```diff
@@ -0,0 +1,113 @@
+      </field>
+    </record>
+
+    <record id="view_vat_listing_tree" model="ir.ui.view">
```
optional:
id pattern

> For a view: <model_name>_view_<view_type>, where view_type is kanban, form, tree, search, ...

tbaden commented on l10n_be_vat_reports/wizard/l10n_be_partner_vat_listing.xml:
```diff
@@ -0,0 +1,113 @@
+    </record>
+
+    <!-- osv_memory for vat listing of clients -->
+    <record id="view_vat_listing_form" model="ir.ui.view">
```
optional:
id pattern

> For a view: <model_name>_view_<view_type>, where view_type is kanban, form, tree, search, ...

tbaden commented on l10n_be_vat_reports/wizard/l10n_be_partner_vat_listing.xml:
```diff
@@ -0,0 +1,113 @@
+      </field>
+    </record>
+
+    <record id="view_vat_listing_result" model="ir.ui.view">
```
optional:
id pattern

> For a view: <model_name>_view_<view_type>, where view_type is kanban, form, tree, search, ...

tbaden commented on l10n_be_vat_reports/wizard/l10n_be_partner_vat_listing.xml:
```diff
@@ -0,0 +1,113 @@
+
+    <menuitem name="Annual Listing Of VAT-Subjected Customers" parent="l10n_be.account_reports_be_statements_menu" action="action_partner_vat_listing" id="partner_vat_listing"/>
+
+    <record id="view_vat_listing" model="ir.ui.view">
```
optional:
id pattern

> For a view: <model_name>_view_<view_type>, where view_type is kanban, form, tree, search, ...

tbaden commented on l10n_be_vat_reports/wizard/l10n_be_partner_vat_listing.xml:
```diff
@@ -0,0 +1,113 @@
+      </field>
+    </record>
+
+    <record id="action_partner_vat_listing" model="ir.actions.act_window">
```
optional:
id pattern

> For an action: the main action respects <model_name>_action. Others are suffixed with _<detail>, where detail is an underscore lowercase string explaining the action (should not be long). This is used only if multiple actions are declared for the model.

tbaden commented on l10n_be_vat_reports/wizard/l10n_be_partner_vat_listing.xml:
```diff
@@ -0,0 +1,113 @@
+<openerp>
+  <data>
+
+    <record id="view_partner_vat_listing" model="ir.ui.view">
```
optional:
id pattern

> For a view: <model_name>_view_<view_type>, where view_type is kanban, form, tree, search, ...

tbaden commented on l10n_be_vat_reports/__openerp__.py:
```diff
@@ -0,0 +1,45 @@
+    'name': 'Belgium VAT Reports',
+    'author': 'ACSONE SA/NV,'
+              'Odoo Community Association (OCA)',
+    'website': 'http://acsone.eu',
```
please set website to: https://github.com/OCA/l10n-belgium

tbaden reviewed (changes requested):
thanks for the work 👍 
I did a code review: just some small changes:
please use the new fragment [readme](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-9.0)

robinkeunen commented:
Thanks for the review, @tbaden. I used readme fragments and fixed the website. However, I did not change the xml id as I don't want to risk breaking code using these ids. But thanks for the recommendations, I always wonder how to name the views consistently.

robinkeunen commented:
I also rebased on `oca/9.0`

remytms commented:
@tbaden Is it possible to merge this PR ? :)

houssine78 commented:
@sbidoul I think this PR is ready for merging :) I ping you as it stayed without response after taken the comment into account ci and runbot are green and we use it in production since then.

Thanks and happy new year !

sbidoul commented:
@houssine78 we still miss the correct attribution to Odoo unless I miss somehintg. I commented in #75: let's finish that one first.

Happy new year!

houssine78 committed:
[FIX] fix missing where clause in sql query

without the WHERE clause the conditional clause is applied to the join
which leads to take extra move lines into account

houssine78 committed:
[COPY] add mention to Odoo SA as authors

remytms reviewed (approved):
LGTM

houssine78 commented:
@sbidoul I fixed to correct attribution to odoo and provided a fix. Could you tell me if we're good ?

robinkeunen committed:
[FIX] l10n_be_vat_reports: website key

robinkeunen committed:
[IMP] l10n_be_vat_reports: readme

robinkeunen committed:
[REF] l10n_be_vat_reports: blacken
